cliquet/__init__.py

```diff
@@ -117,6 +117,18 @@ def includeme(config):
     # Public settings registry.
     config.registry.public_settings = {'batch_max_requests', 'readonly'}
 
+    # Directive to declare arbitrary API capabilities.
+    def add_api_capability(config, identifier, description="", url="", **kw):
+        if identifier in config.registry.api_capabilities:
+            error_msg = "API capability '%s' already exists." % identifier
```

Comment:
I think we can be more explicit here: `"The '%s' capability is already defined, with the values %s. Only one plugin can provide one capability" % (identifier, config.registry.api_capabilites[identifier])`.